SConstruct

```diff
@@ -1,83 +1,111 @@
-env = Environment()
-
-# uncomment this to use an OpenXR-SDK/build dir adjacent to godot_openxr
-# env.Append(LIBPATH = ['../OpenXR-SDK/build/src/loader'])
-# env.Append(CPPPATH='../OpenXR-SDK/build/include')
+# Reads variables from an optional file.
+customs = ['../custom.py']
+opts = Variables(customs, ARGUMENTS)
+
+# Gets the standard flags CC, CCX, etc.
+env = DefaultEnvironment()
+
+# Define our parameters
+opts.Add(EnumVariable('platform', "Platform", 'windows', ['linux', 'osx', 'windows']))
+opts.Add(EnumVariable('target', "Compilation target", 'release', ['d', 'debug', 'r', 'release']))
+opts.AddVariables(
+    PathVariable('openxr_include_path', 'The path where the OpenXR include files are.', 'OpenXR-SDK/'),
+    PathVariable('openxr_library_path', 'The path where the OpenXR library files are.', 'OpenXR-SDK/'),
+    PathVariable('target_path', 'The path where the lib is installed.', 'demo/addons/godot-openxr/bin/'),
+    PathVariable('target_name', 'The library name.', 'godot_openxr', PathVariable.PathAccept),
+)
+opts.Add(BoolVariable('use_llvm', "Use the LLVM / Clang compiler", 'no'))
+
+# Updates the environment with the option variables.
+opts.Update(env)
+
+# Paths
+godot_headers_path = "godot_headers/"
+target_path = env['target_path']
+
+default_openxr_include_path = "OpenXR-SDK/build/"
+default_openxr_library_path = "OpenXR-SDK/build/"
+
+if env['platform'] == "windows":
+    target_path += "win64/"
+    default_openxr_include_path += "win64/include/"
+    default_openxr_library_path += "win64/src/loader"
+
+    # Check some environment settings
+    if env['use_llvm']:
+        env['CXX'] = 'clang++'
+
+        if env['target'] in ('debug', 'd'):
+            env.Append(CCFLAGS = ['-fPIC', '-g3','-Og', '-std=c++17'])
+        else:
+            env.Append(CCFLAGS = ['-fPIC', '-g','-O3', '-std=c++17'])
+    else:
+        # This makes sure to keep the session environment variables on windows,
+        # that way you can run scons in a vs 2017 prompt and it will find all the required tools
+        env.Append(ENV = os.environ)
 
-if platform == "windows":
-    env = Environment(ENV = os.environ)
+        env.Append(CCFLAGS = ['-DWIN32', '-D_WIN32', '-D_WINDOWS', '-W3', '-GR', '-D_CRT_SECURE_NO_WARNINGS'])
+        if env['target'] in ('debug', 'd'):
+            env.Append(CCFLAGS = ['-EHsc', '-D_DEBUG', '/MTd'])
+        else:
+            env.Append(CCFLAGS = ['-O2', '-EHsc', '-DNDEBUG', '/MT'])
 
-if ARGUMENTS.get("use_llvm", "no") == "yes":
-    env["CXX"] = "clang++"
+    # Do we need these?
+    env.Append(LIBS=["opengl32", "setupapi", "advapi32.lib"])
 
-def add_sources(sources, directory):
-    for file in os.listdir(directory):
-        if file.endswith('.c'):
-            sources.append(directory + '/' + file)
-        elif file.endswith('.cpp'):
-            sources.append(directory + '/' + file)
+elif env['platform'] == "osx":
+    # not tested
```

Comment:
OpenXR on Mac is not really a thing for now and I suspect the OpenXR SDK itself won't work properly on Mac. For now we only need to concern ourselves with Windows and Linux.

Reply:
I've commented out the OSX one and fixed the typo in Linux :)